tests/driver_pulse_counter/main.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2017 UC Berkeley
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Test application for the PULSE_COUNTER driver
+ *
+ * @author      Hyung-Sin Kim <hs.kim@cs.berkeley.edu>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "xtimer.h"
+#include "saul_reg.h"
+#include "phydat.h"
+
+#define SLEEP       (1000 * 1000U)
+
+saul_reg_t *sensor_pulse_counter_t;
```

Comment:
No need to use saul in the test application but rather the public API of the driver.

SAUL can be used when building the `test/saul` or `examples/saul` applications. But I admit the saul adaption won't be built by default by the CI. See #6405